organization/_posts/2018-02-21-packaging.md

```diff
@@ -0,0 +1,113 @@
+---
+title: "HSF Packaging Group Meeting #19, February 21, 2018"
+layout: default
+---
+
+# {{page.title}}
+
+Agenda
+======
+
+[https://indico.cern.ch/event/704402/](https://indico.cern.ch/event/704402/)
+
+Participants: Ben Morgan, Chris Green, Guilherme Amadio, Jim Amundson,
+Oana Boeriu, Patrick Gartung, Graeme Stewart, Emil Obreshkov, Giulio
+Eulisse, Marco Clemencic, Javier Cervantes Villanueva, Shahzad Malik
+Muzaffar, Geri Ganis
+
+Introduction - Graeme
+=====================
+
+-   Very brief! No updates on Use Cases and TestStack. Graeme and Ben
+    will discuss later this week.
+-   Talk on AliBuild - perspective of "user/developer" rather than
+    "packager"
+-   Open discussion on Spack, using Javier's talk/question from last
+    time as a starting point
+
+AliBuild Experience - Giulio
+============================
+
+-   aliBuild: packaging/build/setup tool for ALICE experiment. Used to
+    build whole stack and prepare user work area/environment.
+-   Aimed at both end user and packager/librarian
+-   Started in July 2015, full implementation/use by end users in May
+    2016
+    -   \~5 months for initial implementation and infrastructure
+    -   Ongoing aggregated 1FTE to improve build recipes, bug fixes,
+        etc.
+-   About 300-500 users (distinct installs)
+-   Even mix of OS/Arch/Versions
+-   Small set of current versions
+-   Right choices:
+    -   Git for all sources, even third party
+    -   Bash for all build recipes, main point is allowing contributions
+        from non-experts
+    -   YAML for metadata
+    -   Conventions over templates: conventions on environment that
+        recipe runs in.
+    -   Tarballs as file formats, rpms/dpkgs in separate step by FPM
+    -   Dependency Management:
+        -   No central "config" - derived only from set of dependencies
+            (Graeme: back propagation? Yes)
```

Comment:
We actually do a topological sort to decide the build order, but the list of edges is created via back propagation.